src/test/compile-fail/wf-static-method.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// check that static methods don't get to assume `Self` is well-formed
```

Comment:
So I've been meaning to get back to you on this test. I'm not quite sure what this is testing... it seems like the `Self` type, if WF, would imply that 'b: 'a, but the various tests are erroring because 'a: 'b doesn't hold. Is this testing what you think it is?

Also, within an impl block, part of RFC 1214 was that we DO get to assume that the Self (and other input types) to the trait are well-formed, which is why we check that the trait ref is WF when we project. This was used to reduce the overhead on associated type values, but I imagine it should apply to static fns as well.